pci/src/bus.rs

```diff
@@ -166,15 +166,42 @@ impl PciBus {
         Ok(())
     }
 
-    pub fn next_device_id(&mut self) -> Result<u32> {
-        for (idx, device_id) in self.device_ids.iter_mut().enumerate() {
-            if !(*device_id) {
-                *device_id = true;
-                return Ok(idx as u32);
+    /// Allocates a PCI device ID on the bus.
+    ///
+    /// - `id`: ID to allocate on the bus. If [`None`], the next free
+    ///         device ID on the bus is allocated, else the ID given is
```

Comment:
This will not pass cargo fmt (or cargo clippy?). `device ID` should start on the same level as the \` from the line above